1-Draft/RFCNNNN-Side-By-Side-Support.md

```diff
@@ -0,0 +1,253 @@
+PowerShell 7 will align more closely with the .NET Core support lifecycle,
+which brings in support for LTS (Long Term Servicing) and non-LTS releases.
+Non-LTS (aka Current) releases are supported for three months after a
+subsequent Current release or an LTS release. This means you'll have to update
+those releases regularly, which helps move the community forward. On the other
+hand, LTS releases will be supported either for three years after general
+availability or for a 12 month maintenance period after the next LTS release
+ships, whichever is longer (i.e. if it takes them longer than expected to get
+an LTS release out the door for some reason, the previous LTS release could be
+supported for even longer). This means you'll be able to stick your heels in
+the ground and use an LTS release for a very long time, which risks holding us
+back.
+
+In order to keep the focus on moving forward, the purpose of this proposal is
+to support side-by-side installations of PowerShell such that...:
+
+* ...it is easy to see what versions of PowerShell are installed.
+* ...it is easy to launch a specific installed version of PowerShell from a
+command line.
+* ...having multiple installed versions of PowerShell doesn't cause problems
+for automation, products, or services built for a specific version of PowerShell.
+
+## Motivation
+
+    As a user,
+    I can easily install newer versions of PowerShell side-by-side with old versions,
+    so that I can continue to use products/solutions for specific versions of PowerShell while using newer versions when needed.
+
+    As a user,
+    I can use the command line to see which versions of PowerShell are installed,
+    so that I can know what versions are available to me to use.
+
+    As a user,
+    I can launch the version of PowerShell I need from the command line,
+    so that I can use automation built and tested against specific versions of PowerShell while still being able to use the latest version on the same system.
+
+## User Experience
+
+### Installing a specific version of PowerShell
+
+The end goal is for users to be easily able to do any of the following with a
+PowerShell installer:
+
+* install the latest version of PowerShell side-by-side with an existing
+version from a UI (e.g. `.msi`, `.pkg`, `.deb`)
+* install the latest version of PowerShell side-by-side with an existing
+version from the command line (e.g. `sudo apt-get install ...`, `brew cask
+install ...`, `choco install ...`)
+
+_Note: I'm a little uncertain of these details here, so please correct me where
+appropriate._
+
+The current release process seems to include updated published versions of
+packages in the various package repositories (for `apt-get`, homebrew, and
+Chocolatey). The problem with this approach is that it does not support
+side-by-side installations at the command line. For example, if I have
+PowerShell Core 6.2.0 installed on macOS and I want to install PowerShell Core
```

Comment:
Is there a real need to have patch versions side-by-side?  I would think the common scenario is 6.2, 7.0, 7.1.

Reply:
Indeed.  And if PowerShell releases truly follow semver rules for versioning, I would expect 7.1 to be completely back compat with 7.0.  So I'd only expect side-by-side support for **major** versions e.g. 5, 6, 7, 8, etc.

Reply:
We're explicitly not using semver for the version of PowerShell itself. Modules should use semver, but PS versions are just intended to signal relative size of feature sets, among other things.